File: ros_ws/src/intro_pkg1/src/NumberSubscriber.cpp
// Include the ROS header file
#include "ros/ros.h"
// Message data type
#include "std_msgs/Float64.h"

// Caller function  
void receivedMessage(const std_msgs::Float64::ConstPtr &msg) {
    ROS_INFO("Number received: %f", msg->data);
}

int main(int argc, char **argv) {
    // Initialize the ROS node
    ros::init(argc, argv, "NumberSubscriber");
    // Node Handler
    ros::NodeHandle nodeHandler;
    // Subscribe to the topic: "floating_numbers"
    ros::Subscriber sub = nodeHandler.subscribe<std_msgs::Float64>("floating_numbers", 2, receivedMessage);
    // Spin node
    ros::spin();
    return 0;
}

File: ros_ws/src/intro_pkg1/scripts/NumberSubscriber.py
#!/usr/bin/env python
import rospy
from std_msgs.msg import Float64

# Function callback
def msgReceived(msg):
    # Parse the received message as Float 64 type
    message = Float64(msg)
    # Log that as INFO
    rospy.loginfo("Received: {0}".format(message.data))

def main():
    # Initialize the node
    rospy.init_node("NumberSubscriber", anonymous=True)
    # Setup the subscription
    subscriberObject = rospy.Subscriber("floating_numbers", Float64, msgReceived)
    # Wait for the messages forever
    rospy.spin()

# Code entrypoint
if __name__ == "__main__":
    try:
        main()
    except rospy.ROSInterruptException:
        rospy.logerr("An error has occurred")
